test/zipObj.js

```diff
@@ -11,9 +11,8 @@ describe('zipObj', function() {
     eq(R.zipObj(['a', 'b', 'c'], [1, 2, 3, 4, 5, 6, 7]), {a: 1, b: 2, c: 3});
   });
 
-  it('extra keys are undefined', function() {
```

Comment:
> Was there a particular reason for supporting undefined values, or did this simply fall out of the implementation?

I don't recall. So it may be "Development driven tasting". 

Some things bug me about this:

* if were to implement a safe version of this, it would guarantee a value for each key; the value would simply be wrapped in a `Maybe`. 
* i'm afraid of generating different inconsistency w/r/t `for-in` tests with this implementation. But as we've discussed before, what exactly is the difference between a key that isn't there and a key that is mapped to `undefined`?